jlm/llvm/ir/operators/call.cpp

```diff
@@ -234,15 +229,22 @@ CallNode::TraceFunctionInput(const CallNode & callNode)
       return origin;
     }
 
-    if (auto phiInputArgument = dynamic_cast<const phi::cvargument *>(origin))
+    if (auto phi = rvsdg::TryGetRegionParentNode<rvsdg::PhiNode>(*origin))
     {
-      origin = phiInputArgument->input()->origin();
-      continue;
+      auto var = phi->MapArgument(*origin);
+      if (auto fix = std::get_if<rvsdg::PhiNode::FixVar>(&var))
+      {
+        return fix->recref;
+      }
+      else if (auto ctx = std::get_if<rvsdg::PhiNode::ContextVar>(&var))
+      {
+        return ctx->input->origin();
+      }
     }
```

Comment:
Could we get an else with JLM_UNREACHABLE here, just to be sure we exhausted all the possibilities

Reply:
added